OpenRA.Game/GameRules/Ruleset.cs

```diff
@@ -82,6 +82,19 @@ public class Ruleset
 						}
 					}
 				}
+
+				var projectileLoaded = weapon.Value.Projectile as IRulesetLoaded<WeaponInfo>;
```

Comment:
`weapon.Value.Projectile` is an `IProjectileInfo` so this will always be null.

Reply:
This is definitely wrong though as-is, since this was the version @ABrandau was constantly crashing with, so this line is _not_ functional. I'm quite certain it should be `weapon.Value as IRulesetLoaded<WeaponInfo>;` then.

There is a testcase in AS/Engine-ABrandau/OpenRA as WarheadTrailProjectile within their AS namespace.

Reply:
```cs
weapon.Value as IRulesetLoaded<WeaponInfo>;
```
Cannot be converted to OpenRA.Traits.IRulesetloaded<OpenRA.GameRules.WeaponInfo>